ParProcCo/job_scheduler.py

```diff
@@ -414,55 +414,56 @@ def fetch_and_update_state(self, job_id: int) -> JobResponseProperties:
         self.update_status_infos(ji)
         return ji
 
-    def wait_all_jobs_terminated(
-        self, job_ids: List[int], check_time: int
+    def wait_all_jobs(
+        self,
+        job_ids: List[int],
+        required_state: STATEGROUP,
+        timeout: Optional[int],
+        sleep_time: Optional[int],
     ) -> list[int]:
-        wait_until = time.time() + check_time
+        ended = required_state == STATEGROUP.ENDED
+        if timeout is None:
+            timeout = 60 if ended else 15
+        if sleep_time is None:
+            sleep_time = 60 if ended else 5
+        wait_until = time.time() + timeout
         remaining_jobs = job_ids.copy()
-        sleep_time = min(60, check_time)
         while len(remaining_jobs) > 0 and time.time() <= wait_until:
             for job_id in list(remaining_jobs):
                 self.fetch_and_update_state(job_id)
-                if self.status_infos[job_id].current_state in STATEGROUP.ENDED:
+                # check if current_state is in STATEGROUP.ENDED or not in STATEGROUP.STARTING
+                if (self.status_infos[job_id].current_state in required_state) is ended:
```

Comment:
`==` instead of `is`? Your comment is inaccurate!

Reply:
Using literals for STATEGROUP now.